machine-runner/src/runner/runner.ts

```diff
@@ -520,10 +527,12 @@ export const createMachineRunnerInternal = <
 
     emitter.on('next', nva.push)
     emitter.on('failure', nva.fail)
+    emitter.on('commandPersisted', nva.purge)
```

Comment:
Ah, this might have been a misunderstanding or I might miss something: my understanding was that
```typescript
someState.commands().someCommand()
```
should directly purge the awaiter state; I didn’t think that we need to wait until persistence is done.

Reply:
Yes, that’s a good point, thanks for the explanation!